libraries/chain/dbs_atomicswap.cpp

```diff
@@ -0,0 +1,129 @@
+#include <scorum/chain/dbs_atomicswap.hpp>
+#include <scorum/chain/database.hpp>
+#include <scorum/chain/dbs_account.hpp>
+#include <scorum/chain/account_object.hpp>
+#include <scorum/protocol/atomicswap_helper.hpp>
+
+using namespace scorum::protocol;
+
+namespace scorum {
+namespace chain {
+
+dbs_atomicswap::dbs_atomicswap(database& db)
+    : _base_type(db)
+{
+}
+
+dbs_atomicswap::atomicswap_contracts_refs_type dbs_atomicswap::get_contracts(const account_object& owner) const
+{
+    atomicswap_contracts_refs_type ret;
+
+    const auto& idx
+        = db_impl().get_index<atomicswap_contract_index>().indicies().get<by_owner_name>().equal_range(owner.name);
+    for (auto it = idx.first; it != idx.second; ++it)
+    {
+        ret.push_back(std::cref(*it));
+    }
+
+    return ret;
+}
+
+const atomicswap_contract_object& dbs_atomicswap::get_contract(const account_object& recipient,
+                                                               const std::string& secret_hash) const
+{
+    FC_ASSERT(!secret_hash.empty(), "Empty secret hash.");
+
+    try
+    {
+        atomicswap::hash_index_type contract_hash = atomicswap::get_contract_hash(recipient.name, secret_hash);
+        return db_impl().get<atomicswap_contract_object, by_contract_hash>(contract_hash);
+    }
+    FC_CAPTURE_AND_RETHROW((recipient.name)(secret_hash))
+}
+
+const atomicswap_contract_object& dbs_atomicswap::create_initiator_contract(const account_object& initiator,
```

Comment:
why ```create_initiator_contract``` not ```create_contract```

Reply:
Because there are diferent 'deadline' and 'close' logic for initiator_contract and participant_contract.